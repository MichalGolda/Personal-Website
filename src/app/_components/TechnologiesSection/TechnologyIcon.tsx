import Image from "next/image";

export type TechnologyIconProps = {
  src: string;
  alt: string;
};

const TechnologyIcon = ({ src, alt }: TechnologyIconProps) => {
  return (
    <Image
      key={src}
      className="max-w-8 max-h-8"
      src={src}
      alt={alt}
      width={32}
      height={32}
    />
  );
};

export const TypescriptIcon = () => (
  <TechnologyIcon src="/typescript.webp" alt="typescript" />
);

export const JavascriptIcon = () => (
  <TechnologyIcon src="/javascript.webp" alt="javascript" />
);

export const ReactIcon = () => <TechnologyIcon src="/react.webp" alt="react" />;

export const VueIcon = () => <TechnologyIcon src="/vue.webp" alt="vue" />;

export const NextJsIcon = () => (
  <TechnologyIcon src="/nextjs.webp" alt="nextjs" />
);

export const NuxtJsIcon = () => (
  <TechnologyIcon src="/nuxtjs.webp" alt="nuxtjs" />
);

export const SassIcon = () => <TechnologyIcon src="/sass.webp" alt="sass" />;

export const TailwindCssIcon = () => (
  <TechnologyIcon src="/tailwindcss.webp" alt="tailwindcss" />
);

export const DjangoIcon = () => (
  <TechnologyIcon src="/django.webp" alt="django" />
);

export const PHPIcon = () => <TechnologyIcon src="/php.webp" alt="php" />;

export const DockerIcon = () => (
  <TechnologyIcon src="/docker.webp" alt="docker" />
);

export const NodeJsIcon = () => (
  <TechnologyIcon src="/nodejs.webp" alt="nodejs" />
);

export const NestJsIcon = () => (
  <TechnologyIcon src="/nestjs.webp" alt="nestjs" />
);

export const PostgreSQLIcon = () => (
  <TechnologyIcon src="/postgresql.webp" alt="postgresql" />
);

export const AWSIcon = () => <TechnologyIcon src="/aws.webp" alt="aws" />;

export const GCPIcon = () => <TechnologyIcon src="/gcp.webp" alt="gcp" />;

export const AzureIcon = () => <TechnologyIcon src="/azure.webp" alt="azure" />;

export const PrettierIcon = () => (
  <TechnologyIcon src="/prettier.webp" alt="prettier" />
);

export const EslintIcon = () => (
  <TechnologyIcon src="/eslint.webp" alt="eslint" />
);

export const FigmaIcon = () => <TechnologyIcon src="/figma.webp" alt="figma" />;

export default TechnologyIcon;
