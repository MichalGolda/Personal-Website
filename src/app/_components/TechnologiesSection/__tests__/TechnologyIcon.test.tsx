import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import TechnologyIcon from "../TechnologyIcon";

describe("TechnologyIcon component", () => {
  it("renders without errors", () => {
    const { container } = render(
      <TechnologyIcon src="/test.webp" alt="test" />
    );

    const image = container.firstChild;

    expect(image).toHaveAttribute(
      "src",
      "/_next/image?url=%2Ftest.webp&w=64&q=75"
    );
    expect(image).toHaveAttribute("alt", "test");
    expect(image).toHaveAttribute("width", "32");
    expect(image).toHaveAttribute("height", "32");
  });

  it("renders unchanged", () => {
    const { container } = render(
      <TechnologyIcon src="/test.webp" alt="test" />
    );

    expect(container).toMatchSnapshot();
  });
});
