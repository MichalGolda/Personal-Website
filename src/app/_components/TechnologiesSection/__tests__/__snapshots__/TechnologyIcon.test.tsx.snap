// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnologyIcon component renders unchanged 1`] = `
<div>
  <img
    alt="test"
    class="max-w-8 max-h-8"
    data-nimg="1"
    decoding="async"
    height="32"
    loading="lazy"
    src="/_next/image?url=%2Ftest.webp&w=64&q=75"
    srcset="/_next/image?url=%2Ftest.webp&w=32&q=75 1x, /_next/image?url=%2Ftest.webp&w=64&q=75 2x"
    style="color: transparent;"
    width="32"
  />
</div>
`;
